Add contributing guidelines Initial discussion (#3)
RISC-OS-Community/DME-Core: .github/PULL_REQUEST_TEMPLATE.md

<!--
  We ❤️  Pull Requests! :-)

  This template is not a must to use, however it would help us a lot to speed up
  your pull request review if you would be so kind and compile it with your code

>   submission an dyour Pull Request!

Reviewer: ```suggestion
  submission and your Pull Request!
```

  If you remove or skip this template, you'll make the Elders of RISC OS sad and 

>   the mighty King Arthur might rise his ARM against you!!!! you're warned :)

Reviewer: ```suggestion
  the mighty King Arthur might raise his ARM against you!!! You have been warned :)
```

Reviewer: In addition, maybe an exclamation-mark after "warned". I'm not how well received this flippant style will be with the community, but you probably know what sense of humour the would-be contributors have better than I do. ;)

  Pull Request Requirements:
  * Please include tests to illustrate the problem this PR resolves.
  * Please update the documentation in `doc` where necessary
  * Please try to describe your goal with this PR

  Please place an x (no spaces - [x]) in all [ ] that apply.
-->

This PR contains:
- [ ] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?
- [ ] no
- [ ] yes (*bugfixes and features will not be merged without tests*)

Breaking Changes?
- [ ] no

> - [ ] yes (*breaking changes will not be merged unless absolutely necessary*)

Reviewer: ```suggestion
- [ ] yes (*breaking changes will not be merged unless absolutely necessary – please provide motivation*)
```

List any relevant issue numbers:

> 

Reviewer: ```suggestion
<!--
  If your PR resolves an issue you can write for instance `Resolves #42` or `Fixes #42` to automatically close that issue when the PR is merged.
-->
```

### Description

<!--
  Please be thorough and clearly explain the problem being solved.
  * If this PR adds a feature, look for previous discussion on the feature by searching the issues first.

>   * Is this PR related to an issue?

Reviewer: ```suggestion
  * Is this PR related to an issue?
  * Does this PR implement any changes that are breaking/not backwards compatible?
```
